components/brave_new_tab_ui/components/default/bitcoinDotCom/index.tsx

```diff
@@ -0,0 +1,310 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+import createWidget from '../widget/index'
+import * as Styled from './style'
+import { StyledTitleTab } from '../widgetTitleTab'
+import assetIcons from './assets/icons'
+import fiatData from './fiat'
+import BitcoinDotComLogo from './assets/logo.png'
+import { CaratDownIcon } from 'brave-ui/components/icons'
+import { getLocale } from '../../../../common/locale'
+
+interface State {
+  selectedAsset: string
+  selectedFiat: string
+  currentAmount: string
+  assetsShowing: boolean
+  fiatCurrenciesShowing: boolean
+}
+
+interface Props {
+  showContent: boolean
+  stackPosition: number
+  onShowContent: () => void
+  onBuyCrypto: () => void
+  onInteraction: () => void
+}
+
+class BitcoinDotCom extends React.PureComponent<Props, State> {
+  private fiatCurrencies: Record<string, string> = fiatData
+  private assets: Record<string, string> = {
+    'BCH': 'Bitcoin Cash',
+    'BTC': 'Bitcoin',
+    'ETH': 'Ethereum',
+    'LTC': 'Litecoin',
+    'XLM': 'Stellar',
+    'XRP': 'Ripple'
+  }
+
+  constructor (props: Props) {
+    super(props)
+    this.state = {
+      selectedAsset: 'BCH',
+      selectedFiat: 'EUR',
+      currentAmount: '',
+      assetsShowing: false,
+      fiatCurrenciesShowing: false
+    }
+  }
+
+  renderTitle () {
+    const { showContent } = this.props
+
+    return (
+      <Styled.Header>
+        <Styled.Title contentShowing={showContent}>
+          <Styled.BitcoinDotComIcon>
+            <img src={BitcoinDotComLogo} />
+          </Styled.BitcoinDotComIcon>
+          <Styled.TitleText>
+            {'Bitcoin.com'}
+          </Styled.TitleText>
+        </Styled.Title>
+      </Styled.Header>
+    )
+  }
+
+  toggleAssetsShowing = () => {
+    this.setState({
+      assetsShowing: !this.state.assetsShowing
+    })
+    this.props.onInteraction()
+  }
+
+  setSelectedAsset = (asset: string) => {
+    this.setState({
+      assetsShowing: false,
+      selectedAsset: asset
+    })
+  }
+
+  setCurrentAmount = (event: any) => {
+    const { target } = event
+    const { value } = target
+
+    this.props.onInteraction()
+
+    if (isNaN(parseFloat(value))) {
```

Comment:
This won't work to validate or do anything really. `parseFloat` strips out any non-numeric characters, so `isNaN` of that output will always return `false` and then since you aren't using `parseFloat` for the actual saved value, it persists the incorrect value.
```js
parseFloat('2dd') // -> 2
isNaN(2) // -> false
```

So perhaps you want to simply strip the non-numeric from the input, then short of a regex, just call something like
```js
let transformed = parseFloat(value)
transformed = transformed === isNaN ? '' : transformed
this.setState({ currentAmount: transformed })
```

Or if you want to block changes as your code is doing now then
```js
if (isNaN(value)) {
  event.preventDefault()
  return
}
```

All depends whether you want to clean up bad pasted data, or just block bad input entirely.

Reply:
Added regex validation, this will give the most accuracy and block bad input entirely.

Reply:
tested and validation now works